tianshou/data/collector.py

```diff
@@ -140,42 +196,36 @@ def reset(
         reset_buffer: bool = True,
         gym_reset_kwargs: dict[str, Any] | None = None,
     ) -> None:
-        """Reset the environment, statistics, current data and possibly replay memory.
+        """Reset the environment, counters, data and, if desired, the buffer.
 
         :param reset_buffer: if true, reset the replay buffer that is attached
             to the collector.
         :param gym_reset_kwargs: extra keyword arguments to pass into the environment's
-            reset function. Defaults to None (extra keyword arguments)
+            reset function.
         """
-        # use empty Batch for "state" so that self.data supports slicing
-        # convert empty Batch to None when passing data to policy
-        data = Batch(
-            obs={},
-            act={},
-            rew={},
-            terminated={},
-            truncated={},
-            done={},
-            obs_next={},
-            info={},
-            policy={},
-        )
-        self.data = cast(RolloutBatchProtocol, data)
+        self._reset_data()
         self.reset_env(gym_reset_kwargs)
         if reset_buffer:
             self.reset_buffer()
-        self.reset_stat()
+        self.reset_counters()
+
+    def _reset_data(self) -> None:
+        self.data = get_empty_rollout_batch()
 
-    def reset_stat(self) -> None:
-        """Reset the statistic variables."""
+    def reset_counters(self) -> None:
+        """Reset the collection counting fields."""
         self.collect_step, self.collect_episode, self.collect_time = 0, 0, 0.0
 
     def reset_buffer(self, keep_statistics: bool = False) -> None:
         """Reset the data buffer."""
         self.buffer.reset(keep_statistics=keep_statistics)
 
-    def reset_env(self, gym_reset_kwargs: dict[str, Any] | None = None) -> None:
-        """Reset all of the environments."""
+    def reset_env(
+        self,
+        gym_reset_kwargs: dict[str, Any] | None = None,
+        set_obs_next_to_obs: bool = False,
```

Comment:
This is a non-trivial functionality, why is it needed and when would one want to use it? Pls add a docstring